NonSucking.Framework.Serialization/GeneratedSerializerCode.cs

```diff
@@ -0,0 +1,151 @@
+
+    public IEnumerable<StatementSyntax> ToMergedBlock()
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            yield return declaration.ToDeclarationAndAssignment();
+        }
+
+        foreach (var statement in Statements)
+        {
+            yield return statement;
+        }
+    }
+
+    public IEnumerable<StatementSyntax> MergeBlocksSeperated(GeneratedSerializerCode other,
+        Func<SerializerVariable, LocalDeclarationStatementSyntax> variableTransformer = null)
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            var newDeclaration = variableTransformer == null
+                                    ? declaration.Declaration
+                                    : variableTransformer(declaration);
+            
+            other.VariableDeclarations.Add(new SerializerVariable(newDeclaration, declaration.OriginalMember, declaration.UniqueName, null));
+        }
+        return VariableDeclarations.Where(x => x.InitialValue != null).Select(x => x.GetAssignment()).Concat(Statements);
+    }
+
+    public void MergeWith(GeneratedSerializerCode other, bool emitVariables = true)
+    {
+        if (Statements.Count == 0)
+        {
+            foreach (var d in other.VariableDeclarations)
+                VariableDeclarations.Add(d);
+        }
+        else if (emitVariables)
+        {
+            foreach (var d in other.VariableDeclarations)
+            {
+                VariableDeclarations.Add(new SerializerVariable(d.Declaration, d.OriginalMember, d.UniqueName, null));
+                Statements.Add(d.GetAssignment());
+            }
+        }
+        else
+        {
+            foreach (var d in other.VariableDeclarations)
+            {
+                Statements.Add(d.ToDeclarationAndAssignment());
+            }
+        }
+        foreach (var s in other.Statements)
+            Statements.Add(s);
+    }
+    
+    
+    public void DeclareAndAssign(MemberInfo member, string memberName, ITypeSymbol type, ExpressionSyntax valueExpression)
+    {
+        var typeSyntax = SyntaxFactory.ParseTypeName(type.ToDisplayString()); // TODO better type handling?
+        DeclareAndAssign(member, memberName, typeSyntax, valueExpression);
+    }
+    public void DeclareAndAssign(MemberInfo member, string memberName, TypeSyntax typeSyntax, ExpressionSyntax valueExpression)
+    {
+        var variableDeclaration = Statement.Declaration.Declare(memberName, typeSyntax);
+        if (Statements.Count == 0)
+        {
+            VariableDeclarations.Add(
+                new SerializerVariable(variableDeclaration, member, memberName,
+                        valueExpression == null ? null : SyntaxFactory.EqualsValueClause(valueExpression)
+                    )
+                );
+        }
+        else
+        {
+            VariableDeclarations.Add( new SerializerVariable(variableDeclaration, member, memberName, null));
```

Comment:
Doesn't rider remove those spaces?
```suggestion
            VariableDeclarations.Add(new SerializerVariable(variableDeclaration, member, memberName, null));
```